src/lib/ContractParser.js

```diff
@@ -147,32 +158,70 @@ export class ContractParser {
     }, {})
   }
 
-  hasMethodSelector (txInputData, selector) {
-    return (selector && txInputData) ? txInputData.includes(selector) : null
-  }
+  
 
   getMethodsBySelectors (txInputData) {
     let methods = this.getMethodsSelectors()
     return Object.keys(methods)
-      .filter(method => this.hasMethodSelector(txInputData, methods[method]) === true)
+      .filter(method => hasMethodSelector(txInputData, methods[method]) === true)
   }
 
   async getContractInfo (txInputData, contract) {
+    let { interfaces, methods } = await this.getContractImplementedInterfaces(txInputData, contract)
+
+    interfaces = mapInterfacesToERCs(interfaces)
+    return { methods, interfaces }
+  }
+
+  async getContractInfoIfProxy (contractAddress) {
+    const { isUpgradeable, impContractAddress } = await this.checkForUpgradeableContract(contractAddress)
+    if (isUpgradeable) {
+      // manual check required
+      const proxyContractBytecode = await this.getContractCodeFromNode(impContractAddress)
+      const methods = this.getMethodsBySelectors(proxyContractBytecode)
+      let interfaces = this.getInterfacesByMethods(methods)
+
+      interfaces = mapInterfacesToERCs(interfaces)
+      return { methods, interfaces }
```

Comment:
Instead of returning the traditional info, I would return the info of the eip 1967 with its particular format. Then display it in the screen as a _proxy info_, rather than hiding the real contract info.

Reply:
I like it. I intend for this changes to be minimum. I will track this suggestion in the board